mp/src/game/server/momentum/mom_lobby_system.cpp

```diff
@@ -750,126 +810,128 @@ void CMomentumLobbySystem::SendAndReceiveP2PPackets()
     if (m_mapLobbyGhosts.Count() == 0)
         return;
 
-    uint32 size;
-    while (SteamNetworking()->IsP2PPacketAvailable(&size))
+    CHECK_STEAM_API(SteamNetworkingMessages());
+
+    SteamNetworkingMessage_t *messages[64];
+    int read = SteamNetworkingMessages()->ReceiveMessagesOnChannel(0, messages, 64);
```

Comment:
If we want to maintain the old behavior this should be a loop that keeps pumping out the messages until there aren't any left. Otherwise, any leftover messages will be left until the next time this function is visited.

Don't think it's likely that there will be more than 64 messages at once, but still something worth considering.

Reply:
I did doubt the same at first but if, theoretically, somebody joins a server nearing the Steam member limit of 255, you'd be getting roughly 40 * 255 packets per second, and that's just position updates.

It'd be interesting to actually test this, but still, even with 20 people you'd be getting roughly 800 packets per second, and this is run every tick, so doing 64 message reads * 66.6666 times per second should cover it. I'll write a while-loop to read all on the channel in 64-message-chunks and we should be good.

Reply:
Resolved in https://github.com/momentum-mod/game/pull/1000/commits/6290a570f711197a7f9beb6f0bd5f8ae9194f777